client/lib/query-manager/index.js

```diff
@@ -104,6 +104,27 @@ export default class QueryManager {
 	}
 
 	/**
+	 * A sorting function that defines the sort order of items under
+	 * consideration of the specified query.
+	 *
+	 * @param  {Array}  keys  Keys to be sorted
+	 * @param  {Array}  items Items by which to sort
+	 * @param  {Object} query Query object
+	 * @return {Array}        Sorted keys
+	 */
+	sort( keys, items, query ) {
```

Comment:
Yeah, this is kinda ugly. But since it's the keys we're sorting based on information found in the items, I couldn't think of a more elegant way. (I actually did spend a bit of time thinking about this, too. I might not be seeing the forest for the trees tho 😄).